python_modules/libraries/dagster-dbt/dagster_dbt/cli/app.py

```diff
@@ -0,0 +1,140 @@
+import shutil
+import subprocess
+from pathlib import Path
+
+import typer
+from rich.console import Console
+from rich.markdown import Markdown
+from typing_extensions import Annotated
+
+from ..include import STARTER_PROJECT_PATH
+from ..version import __version__
+
+app = typer.Typer(no_args_is_help=True)
+project_app = typer.Typer(no_args_is_help=True)
+app.add_typer(
+    project_app,
+    name="project",
+    help="Commands to initialize a new Dagster project with dbt.",
+)
+
+DBT_PROJECT_YML_NAME = "dbt_project.yml"
+DAGSTER_DBT_PROJECT_DIR_NAME = "dagster"
+
+
+def validate_dagster_project_name(project_name: str) -> str:
+    if not project_name.isidentifier():
+        raise typer.BadParameter(
+            "The project name must be a valid Python identifier containing only letters, digits, or"
+            " underscores."
+        )
+
+    return project_name
+
+
+def validate_dbt_project_dir(dbt_project_dir: Path) -> Path:
+    dbt_project_yaml_path = dbt_project_dir.joinpath(DBT_PROJECT_YML_NAME)
+
+    if not dbt_project_yaml_path.exists():
+        raise typer.BadParameter(
+            f"{dbt_project_dir} does not contain a {DBT_PROJECT_YML_NAME} file. Please specify a"
+            " valid path to a dbt project."
+        )
+
+    return dbt_project_dir
+
+
+@project_app.command(name="scaffold")
+def project_scaffold_command(
+    project_name: Annotated[
+        str,
+        typer.Option(
+            default=...,
+            show_default=False,
+            help="The name of the Dagster project to scaffold.",
+            prompt="Enter a name for your Dagster project (letters, digits, underscores)",
+            callback=validate_dagster_project_name,
+        ),
+    ],
+    dbt_project_dir: Annotated[
+        Path,
+        typer.Option(
+            default=...,
+            help=(
+                "The path of your dbt project directory. By default, we use the current"
+                " working directory."
+            ),
+            callback=validate_dbt_project_dir,
+            exists=True,
+            file_okay=False,
+            dir_okay=True,
+            writable=True,
+            resolve_path=True,
+        ),
+    ] = Path.cwd(),
+    generate_dbt_manifest: Annotated[
+        bool,
+        typer.Option(
+            default=...,
+            help="Create the dbt manifest.",
+        ),
+    ] = True,
+) -> None:
+    """Scaffold a new Dagster project with an existing dbt project."""
```

Comment:
Does this get displayed to the user? If so, I think it would be beneficial to use more specific wording than "with". And it might not be clear what "scaffold" means.

Perhaps "Create files and folders for a new Dagster project that loads assets from an existing dbt project."